### PR TITLE
Added the updated feature_description.mdx file considering all the reviews in my previous reviews

### PR DESCRIPTION
**Pull request by aman06012003**

### CODE REVIEW

**Comment by aman06012003** [comment]
Hi @merveenoyan @MKhalusova @johko I have updated my feature discription outline based on the inputs I got from my past pr.

**Review by ATaylorAerospace — APPROVED** [review]
This is a great update @aman06012003

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+10. **Image features**
+    - Pixel Values: Strong pixel values matrices or multi-dimensional arrays.
+    - Convolutional Neural Network features(CNN): Extracting features using pre-trained CNN models.
+
+## What Makes A Good Descriptor
+
+A good descriptor in the context of image processing or computer vision is a set of characteristics or features that effectively represent key information about an object or scene in an image. Here are some aspects that contribute to making a good descriptor:
+
+1. **Invariant to transformation:** Descriptors should ideally be robust to variations like rotation, translation, scaling and changes in illumination. This means that regardless of how an object is positioned or how the image is altered, the descriptor should remain relatively unchanged and contain the same description as the original one.
+2. **Distinctivness:** A good descriptor captures unique information about the object. It should be able to discriminate between different objects or parts of image and be distinct enough to differentiate them from similar elements.
```
Comment (ricard-inho) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:25]:
```suggestion
2. **Distinctiveness:** A good descriptor captures unique information about the object. It should be able to discriminate between different objects or parts of an image and be distinct enough to differentiate them from similar elements.

```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+7. **Computational Efficiency:** Efficiency is really a crucial part of a descriptor, especially in real-time applications. Descriptors should be computationally feasible for quick processing, particularly in scenarios where speed is crucial, such as robotics or autonomous vehicles.
+8. **Adaptability:** Descriptors that can adapt or learn from the data they are processing can be highly effective, especially in situations where the characteristics of the objects or scenes may change over time. This can increase the useability of the descriptor.
+9. **Noise Robustness:** Descriptors should be able to handle noise in the image data without significantly compromising their ability to accurately represent the underlying features.
+
+## Some Of The Techniques Used In Feature Descriptors
+
+### SIFT : &#x20;
+![Basic Working of SIFT](https://www.researchgate.net/publication/220242450/figure/fig4/AS:669020810387456@1536518399011/Original-SIFT-algorithm-flow.png)
+
+It stands for scale invariant feature transform. It is widely used algorithm in computer vision and image processing for detecting and describing local features in images.
```
Comment (ricard-inho) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:39]:
```suggestion
It stands for scale invariant feature transform. It is a widely used algorithm in computer vision and image processing for detecting and describing local features in images.

```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+2. **Keypoint Localization:** Once potential keypoints are identified, SIFT refines their positions to sub-pixel accuracy and discards low-contrast key points and keypoints on edges to ensure accurate localization.\*\*
+3. **Orientation Assignment:** SIFT computes a dominant orientation for each keypoint based on local image gradient directions. This step makes the descriptor invariant to image rotation.
+4. **Descriptor Generation:** A descriptor is computed for each keypoint region, capturing information about the local image gradients in the vicinity of the keypoint. This descriptor is a compact representation that encapsulates the key characteristics of the image patch surrounding the keypoint.
+5. **Descriptor Matching:** Finally, these descriptors are used for matching keypoints between different images. The descriptors from one image are compared to those in another image to find correspondences.
+
+You can learn more about SIFT using following references:
```
Comment (ricard-inho) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:49]:
```suggestion
You can learn more about SIFT using the following references:

```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+1. https://docs.opencv.org/4.x/da/df5/tutorial_py_sift_intro.html
+2. https://www.educative.io/answers/what-is-sift
+3. https://www.cse.iitb.ac.in/~ajitvr/CS763/SIFT.pdf
+
+SIFT's robustness to various image transformations and its ability to find distinctive features in an image make it valuable in applications like object recognition, image stitching, and 3D reconstruction.
```
Comment (ricard-inho) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:54]:
```suggestion
SIFT's robustness to various image transformations and its ability to find distinctive features in an image makes it valuable in applications like object recognition, image stitching, and 3D reconstruction.

```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+### 2. SURF: &#x20;
+![Basic Working of SURF](https://www.researchgate.net/publication/318565053/figure/fig2/AS:614214838263810@1523451637097/Flow-Chart-for-SURF-Feature-Detection.png)
+
+
+It stands for Speeded Up Robust features. It is another popular algorithm in computer vision and image processing. It is particularly known for its speed and robustness in detecting and describing local image features.
+
+The basic workflow of SURF is given below:
+
+1. **Integral images:** SURF utilizes integral images, which are precomputed representations of the original image. They allow for fast calculations of rectangular area sums within an image, enabling a quicker feature computation.
+2. **Blob detection:** Similar to other feature detection algorithms , SURF starts by identifying potential interest points or keypoints in the image. It uses a Hessian matrix to detect blobs or regions that exhibit significant variations in intensity in multiple directions and scales. These regions are potential keypoints.
```
Comment (ricard-inho) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:65]:
```suggestion
2. **Blob detection:** Similar to other feature detection algorithms, SURF starts by identifying potential interest points or keypoints in the image. It uses a Hessian matrix to detect blobs or regions that exhibit significant variations in intensity in multiple directions and scales. These regions are potential keypoints.

```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+3. **Scale Selection**: It determine the scale of the key points by identifying regions with significant changes in scale-space. It analyzes the determinant of the Hessian matrix across different scales to find robust keypoints at multiple scales.
```
Comment (ricard-inho) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:66]:
```suggestion
3. **Scale Selection**: It determines the scale of the keypoints by identifying regions with significant changes in scale-space. It analyzes the determinant of the Hessian matrix across different scales to find robust keypoints at multiple scales.

```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+### SIFT : &#x20;
+![Basic Working of SIFT](https://www.researchgate.net/publication/220242450/figure/fig4/AS:669020810387456@1536518399011/Original-SIFT-algorithm-flow.png)
+
+It stands for scale invariant feature transform. It is widely used algorithm in computer vision and image processing for detecting and describing local features in images.
+
+The working of SIFT is given below:
+
+1. **Scale Space Extrema detection:** It starts by detecting potential interest points in an image across multiple scales. It looks for locations in the image where the difference of Gaussian function reaches a maximum or minimum over space and scale. These keypoint locations are considered stable under various scale changes.
+2. **Keypoint Localization:** Once potential keypoints are identified, SIFT refines their positions to sub-pixel accuracy and discards low-contrast key points and keypoints on edges to ensure accurate localization.\*\*
```
Comment (ricard-inho) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:44]:
```suggestion
2. **Keypoint Localization:** Once potential keypoints are identified, SIFT refines their positions to sub-pixel accuracy and discards low-contrast keypoints and keypoints on edges to ensure accurate localization.\*\*

```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+3. **Scale Selection**: It determine the scale of the key points by identifying regions with significant changes in scale-space. It analyzes the determinant of the Hessian matrix across different scales to find robust keypoints at multiple scales.
+4. **Orientation Assignment:** For each detected keypoint, SURF assigns a dominant orientation. This is done by calculating Haar wavelet responses in different directions around the keypoint’s neighborhood. Unlike SIFT, SURF uses a set of rectangular filters (Haar wavelets) applied to subregions of the keypoint's neighborhood. The responses of these filters are used to create a feature vector representing the keypoint.
+5. **Descriptor Matching:** The generated descriptors are then used for matching keypoints between different images. Matching involves comparing the feature vectors of keypoints in one image to those in another image to find correspondences.
+
+You can learn more about SURF using following references:
```
Comment (ricard-inho) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:70]:
```suggestion
You can learn more about SURF using the following references:

```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+1. https://medium.com/@deepanshut041/introduction-to-surf-speeded-up-robust-features-c7396d6e7c4e
+2. https://docs.opencv.org/3.4/df/dd2/tutorial_py_surf_intro.html
+3. https://www.ijtra.com/view/feature-extraction-using-surf-algorithm-for-object-recognition.pdf
+
+The key strengths of SURF lie in its computational efficiency, achieved through the use of integral images and Haar wavelet approximations, while maintaining robustness to scale, rotation, and illumination changes. This makes SURF suitable for reat time applications where speed plays a crucial part in object detection, tracking and image stiching.
```
Comment (ricard-inho) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:75]:
```suggestion
The key strengths of SURF lie in its computational efficiency, achieved through the use of integral images and Haar wavelet approximations while maintaining robustness to scale, rotation, and illumination changes. This makes SURF suitable for real time applications where speed plays a crucial part in object detection, tracking, and image stitching.

```

**Review by ricard-inho — COMMENTED** [review]
@aman06012003 Do you have the link to the previous PR? Main suggestion is to put "keypoint" the same all throw the text. I think either "key points" or "keypoints" should be fine.

**Comment by aman06012003** [comment]
> @aman06012003 Do you have the link to the previous PR? Main suggestion is to put "keypoint" the same all throw the text. I think either "key points" or "keypoints" should be fine.

I have closed that pr actually

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 10]
```diff
@@ -0,0 +1,84 @@
+﻿# Feature Description
+
+## How Can We Represent Features In Data Structures?
+
+Representing features in data structures is crucial for organizing and manipulating data effectively. Features, also known as attributes or variables, can be diverse, ranging from numerical values and categories to more complex structures like images or text. Here are some ways to represent features in data structures:
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:5]:
```suggestion
Representing features in data is crucial for organizing and manipulating data effectively. Features, or attributes or variables, can be diverse, ranging from numerical values and categories to more complex structures like images or text. Here are some ways to represent features in data:

```
Data structure is rather a term for ways to represent a given data to improve complexity in processing/search or memory

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+    - Pixel Values: Strong pixel values matrices or multi-dimensional arrays.
+    - Convolutional Neural Network features(CNN): Extracting features using pre-trained CNN models.
+
+## What Makes A Good Descriptor
+
+A good descriptor in the context of image processing or computer vision is a set of characteristics or features that effectively represent key information about an object or scene in an image. Here are some aspects that contribute to making a good descriptor:
+
+1. **Invariant to transformation:** Descriptors should ideally be robust to variations like rotation, translation, scaling and changes in illumination. This means that regardless of how an object is positioned or how the image is altered, the descriptor should remain relatively unchanged and contain the same description as the original one.
+2. **Distinctiveness:** A good descriptor captures unique information about the object. It should be able to discriminate between different objects or parts of an image and be distinct enough to differentiate them from similar elements.
+
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:26]:
```suggestion
```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+6. **Compatibility with Matching Algorithms:**  Descriptors are often used in conjunction with matching algorithms to find correspondences between different images. A good descriptor should be suitable for the matching algorithm being used, whether it's based on distance metrics, machine learning models, or other techniques.
+7. **Computational Efficiency:** Efficiency is really a crucial part of a descriptor, especially in real-time applications. Descriptors should be computationally feasible for quick processing, particularly in scenarios where speed is crucial, such as robotics or autonomous vehicles.
+8. **Adaptability:** Descriptors that can adapt or learn from the data they are processing can be highly effective, especially in situations where the characteristics of the objects or scenes may change over time. This can increase the useability of the descriptor.
+9. **Noise Robustness:** Descriptors should be able to handle noise in the image data without significantly compromising their ability to accurately represent the underlying features.
+
+## Some Of The Techniques Used In Feature Descriptors
+
+### SIFT : &#x20;
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:37]:
```suggestion
### SIFT 
```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+![Basic Working of SIFT](https://www.researchgate.net/publication/220242450/figure/fig4/AS:669020810387456@1536518399011/Original-SIFT-algorithm-flow.png)
+
+It stands for scale invariant feature transform. It is a widely used algorithm in computer vision and image processing for detecting and describing local features in images.
+
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:41]:
```suggestion
```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+
+The working of SIFT is given below:
+
+1. **Scale Space Extrema detection:** It starts by detecting potential interest points in an image across multiple scales. It looks for locations in the image where the difference of Gaussian function reaches a maximum or minimum over space and scale. These keypoint locations are considered stable under various scale changes.
+2. **Keypoint Localization:** Once potential keypoints are identified, SIFT refines their positions to sub-pixel accuracy and discards low-contrast keypoints and keypoints on edges to ensure accurate localization.\*\*
+
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:47]:
```suggestion
```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+### SIFT : &#x20;
+![Basic Working of SIFT](https://www.researchgate.net/publication/220242450/figure/fig4/AS:669020810387456@1536518399011/Original-SIFT-algorithm-flow.png)
+
+It stands for scale invariant feature transform. It is a widely used algorithm in computer vision and image processing for detecting and describing local features in images.
+
+
+The working of SIFT is given below:
+
+1. **Scale Space Extrema detection:** It starts by detecting potential interest points in an image across multiple scales. It looks for locations in the image where the difference of Gaussian function reaches a maximum or minimum over space and scale. These keypoint locations are considered stable under various scale changes.
+2. **Keypoint Localization:** Once potential keypoints are identified, SIFT refines their positions to sub-pixel accuracy and discards low-contrast keypoints and keypoints on edges to ensure accurate localization.\*\*
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:46]:
```suggestion
2. **Keypoint Localization:** Once potential keypoints are identified, SIFT refines their positions to sub-pixel accuracy and discards low-contrast keypoints and keypoints on edges to ensure accurate localization.
```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+
+3. **Orientation Assignment:** SIFT computes a dominant orientation for each keypoint based on local image gradient directions. This step makes the descriptor invariant to image rotation.
+4. **Descriptor Generation:** A descriptor is computed for each keypoint region, capturing information about the local image gradients in the vicinity of the keypoint. This descriptor is a compact representation that encapsulates the key characteristics of the image patch surrounding the keypoint.
+5. **Descriptor Matching:** Finally, these descriptors are used for matching keypoints between different images. The descriptors from one image are compared to those in another image to find correspondences.
+
+You can learn more about SIFT using the following references:
+
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:53]:
```suggestion
```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+1. https://docs.opencv.org/4.x/da/df5/tutorial_py_sift_intro.html
+2. https://www.educative.io/answers/what-is-sift
+3. https://www.cse.iitb.ac.in/~ajitvr/CS763/SIFT.pdf
+
+SIFT's robustness to various image transformations and its ability to find distinctive features in an image makes it valuable in applications like object recognition, image stitching, and 3D reconstruction.
+
+
+### 2. SURF: &#x20;
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:61]:
```suggestion
### SURF
```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+![Basic Working of SURF](https://www.researchgate.net/publication/318565053/figure/fig2/AS:614214838263810@1523451637097/Flow-Chart-for-SURF-Feature-Detection.png)
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:62]: You can upload the images to hf-vision/course-assets and then load from there

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 19]
```diff
@@ -0,0 +1,84 @@
...
+
+
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:64]:
```suggestion
```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 20]
```diff
@@ -0,0 +1,84 @@
...
+5. **Descriptor Matching:** The generated descriptors are then used for matching keypoints between different images. Matching involves comparing the feature vectors of keypoints in one image to those in another image to find correspondences.
+
+You can learn more about SURF using the following references:
+
+1. https://medium.com/@deepanshut041/introduction-to-surf-speeded-up-robust-features-c7396d6e7c4e
+2. https://docs.opencv.org/3.4/df/dd2/tutorial_py_surf_intro.html
+3. https://www.ijtra.com/view/feature-extraction-using-surf-algorithm-for-object-recognition.pdf
+
+The key strengths of SURF lie in its computational efficiency, achieved through the use of integral images and Haar wavelet approximations while maintaining robustness to scale, rotation, and illumination changes. This makes SURF suitable for real time applications where speed plays a crucial part in object detection, tracking, and image stitching.
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:83]: Maybe you can have this above the references

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 21]
```diff
@@ -0,0 +1,84 @@
...
+2. **Keypoint Localization:** Once potential keypoints are identified, SIFT refines their positions to sub-pixel accuracy and discards low-contrast keypoints and keypoints on edges to ensure accurate localization.\*\*
+
+3. **Orientation Assignment:** SIFT computes a dominant orientation for each keypoint based on local image gradient directions. This step makes the descriptor invariant to image rotation.
+4. **Descriptor Generation:** A descriptor is computed for each keypoint region, capturing information about the local image gradients in the vicinity of the keypoint. This descriptor is a compact representation that encapsulates the key characteristics of the image patch surrounding the keypoint.
+5. **Descriptor Matching:** Finally, these descriptors are used for matching keypoints between different images. The descriptors from one image are compared to those in another image to find correspondences.
+
+You can learn more about SIFT using the following references:
+
+1. https://docs.opencv.org/4.x/da/df5/tutorial_py_sift_intro.html
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:54]: Maybe you can have the references as unnumbered list at the bottom not to break the readability

**Review by merveenoyan — COMMENTED** [review]
I gave mostly format related comments, thanks a lot!

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx (hunk 18 above)
Comment (aman06012003) [reply chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:60]: ok🙂

**Review by aman06012003 — COMMENTED** [review]
Updated the changes

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 22]
```diff
@@ -0,0 +1,83 @@
+﻿# Feature Description
+
+## How Can We Represent Features In Data Structures?
+
+Representing features in data is crucial for organizing and manipulating data effectively. Features, or attributes or variables, can be diverse, ranging from numerical values and categories to more complex structures like images or text. Here are some ways to represent features in data:
+
+
+1. **Numerical features**
+    - Arrays/Lists: Simplest form to store numerical values. Each element in the array corresponds to a feature.
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:9]: You can remove the indentation

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 23]
```diff
@@ -0,0 +1,83 @@
...
+
+## How Can We Represent Features In Data Structures?
+
+Representing features in data is crucial for organizing and manipulating data effectively. Features, or attributes or variables, can be diverse, ranging from numerical values and categories to more complex structures like images or text. Here are some ways to represent features in data:
+
+
+1. **Numerical features**
+    - Arrays/Lists: Simplest form to store numerical values. Each element in the array corresponds to a feature.
+    - Tensors: Multidimensional arrays often used in the machine learning frameworks to handle large sets of numerical data efficiently.
+4. **Categorical features**
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:11]:
```suggestion
2. **Categorical features**

```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 24]
```diff
@@ -0,0 +1,83 @@
...
+    - Dictionaries/Lists: Assigning categories to numerical labels or directly storing categorical values.
+    - One hot encoding:** Transforming categorical variables into binary vectors where each bit represents a category.
+7. **Text features**
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:14]: It's ok if we remove categorical and text features altogether

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 25]
```diff
@@ -0,0 +1,83 @@
...
+    - Bag of Words Representation: Breaking text into individual words and counting their occurrences.
+    - Word Embeddings: Representing words as dense vectors capturing semantic meanings.
+10. **Image features**
+    - Pixel Values: Strong pixel values matrices or multi-dimensional arrays.
+    - Convolutional Neural Network features(CNN): Extracting features using pre-trained CNN models.
+
+## What Makes A Good Descriptor
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:21]:
```suggestion
## What Makes a Good Descriptor

```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 26]
```diff
@@ -0,0 +1,83 @@
...
+
+A good descriptor in the context of image processing or computer vision is a set of characteristics or features that effectively represent key information about an object or scene in an image. Here are some aspects that contribute to making a good descriptor:
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:23]:
```suggestion
A good descriptor in image processing or computer vision is a set of characteristics or features that effectively represent key information about an object or scene in an image. Here are some aspects that contribute to making a good descriptor:

```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 27]
```diff
@@ -0,0 +1,83 @@
...
+6. **Compatibility with Matching Algorithms:**  Descriptors are often used in conjunction with matching algorithms to find correspondences between different images. A good descriptor should be suitable for the matching algorithm being used, whether it's based on distance metrics, machine learning models, or other techniques.
+7. **Computational Efficiency:** Efficiency is really a crucial part of a descriptor, especially in real-time applications. Descriptors should be computationally feasible for quick processing, particularly in scenarios where speed is crucial, such as robotics or autonomous vehicles.
+8. **Adaptability:** Descriptors that can adapt or learn from the data they are processing can be highly effective, especially in situations where the characteristics of the objects or scenes may change over time. This can increase the useability of the descriptor.
+9. **Noise Robustness:** Descriptors should be able to handle noise in the image data without significantly compromising their ability to accurately represent the underlying features.
+
+## Some Of The Techniques Used In Feature Descriptors
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:35]:
```suggestion
## Some of the Techniques Used in Feature Descriptors

```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 28]
```diff
@@ -0,0 +1,83 @@
...
+3. **Orientation Assignment:** SIFT computes a dominant orientation for each keypoint based on local image gradient directions. This step makes the descriptor invariant to image rotation.
+4. **Descriptor Generation:** A descriptor is computed for each keypoint region, capturing information about the local image gradients in the vicinity of the keypoint. This descriptor is a compact representation that encapsulates the key characteristics of the image patch surrounding the keypoint.
+5. **Descriptor Matching:** Finally, these descriptors are used for matching keypoints between different images. The descriptors from one image are compared to those in another image to find correspondences.
+
+SIFT's robustness to various image transformations and its ability to find distinctive features in an image makes it valuable in applications like object recognition, image stitching, and 3D reconstruction.
+
+You can learn more about SIFT using the following references:
+1. https://docs.opencv.org/4.x/da/df5/tutorial_py_sift_intro.html
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:53]: Maybe you can give the links like
```suggestion
1. [Introduction to SIFT (Scale-Invariant Feature Transform)](https://docs.opencv.org/4.x/da/df5/tutorial_py_sift_intro.html)

```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 29]
```diff
@@ -0,0 +1,83 @@
...
+2. https://www.educative.io/answers/what-is-sift
+3. https://www.cse.iitb.ac.in/~ajitvr/CS763/SIFT.pdf
+
+
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:57]:
```suggestion
```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 30]
```diff
@@ -0,0 +1,83 @@
...
+### SURF
+![Basic Working of SURF](https://huggingface.co/datasets/hf-vision/course-assets/blob/main/feature-extraction-feature-matching/Flow-Chart-for-SURF-Feature-Detection.png)
+
+It stands for Speeded Up Robust features. It is another popular algorithm in computer vision and image processing. It is particularly known for its speed and robustness in detecting and describing local image features.
+
+The basic workflow of SURF is given below:
+
+1. **Integral images:** SURF utilizes integral images, which are precomputed representations of the original image. They allow for fast calculations of rectangular area sums within an image, enabling a quicker feature computation.
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:66]:
```suggestion
1. **Integral images:** SURF utilizes integral images, precomputed representations of the original image. They allow fast calculations of rectangular area sums within an image, enabling a quicker feature computation.

```

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 31]
```diff
@@ -0,0 +1,83 @@
...
+
+3. **Scale Selection**: It determines the scale of the keypoints by identifying regions with significant changes in scale-space. It analyzes the determinant of the Hessian matrix across different scales to find robust keypoints at multiple scales.
+
+4. **Orientation Assignment:** For each detected keypoint, SURF assigns a dominant orientation. This is done by calculating Haar wavelet responses in different directions around the keypoint’s neighborhood. Unlike SIFT, SURF uses a set of rectangular filters (Haar wavelets) applied to subregions of the keypoint's neighborhood. The responses of these filters are used to create a feature vector representing the keypoint.
+5. **Descriptor Matching:** The generated descriptors are then used for matching keypoints between different images. Matching involves comparing the feature vectors of keypoints in one image to those in another image to find correspondences.
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:72]: You can put another newline above to be consistent

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx [hunk 32]
```diff
@@ -0,0 +1,83 @@
...
+
+The key strengths of SURF lie in its computational efficiency, achieved through the use of integral images and Haar wavelet approximations while maintaining robustness to scale, rotation, and illumination changes. This makes SURF suitable for real time applications where speed plays a crucial part in object detection, tracking, and image stitching.
```
Comment (merveenoyan) [inline chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:74]:
```suggestion
The key strengths of SURF lie in its computational efficiency, which is achieved through the use of integral images and Haar wavelet approximations while maintaining robustness to scale, rotation, and illumination changes. This makes SURF suitable for real-time applications where speed plays a crucial part in object detection, tracking, and image stitching.

```

**Review by merveenoyan — APPROVED** [review]
I only left nitpicking comments, let's get this merged!!!

**Comment by merveenoyan** [comment]
@aman06012003 Can you fix the (1) in file name btw?

File: chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx (hunk 22 above)
Comment (aman06012003) [reply chapters/en/Unit 1 - Fundamentals/feature_extraction/feature_description (1).mdx:9]: sure

**Comment by aman06012003** [comment]
updated the final changes

**Pull request merged**